src/backend/src/lib.rs

```diff
@@ -147,12 +147,32 @@ fn set_config(arg: InitArg) {
     });
 }
 
+/// Runs housekeeping tasks immediately, then periodically:
+/// - `hourly_housekeeping_tasks`
+fn start_periodic_housekeeping_timers() {
+    // Run housekeeping tasks once, immediately but asynchronously.
+    let immediate = Duration::from_secs(0);
+    set_timer(immediate, || ic_cdk::spawn(hourly_housekeeping_tasks()));
+
+    // Then periodically:
+    let hour = Duration::from_secs(60 * 60);
+    let _ = set_timer_interval(hour, || ic_cdk::spawn(hourly_housekeeping_tasks()));
+}
+
+/// Runs hourly housekeeping tasks:
+/// - Top up the cycles ledger.
+async fn hourly_housekeeping_tasks() {
+    let _ = top_up_cycles_ledger(None).await;
```

Comment:
Should we `ic_cdk::print` the error? Those can be read with canister logs. It can offers us a quick way to debug if necessary. I would maybe not log the result, that would be verbose.

Reply:
Yes, that is worthwhile.  I would like proper stats but something is better than nothing.